src/shared/components/Header/WorkspaceSelectorMenu/WorkspaceSelectorContainer.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { RootState } from '../../../store/reducers';
+import WorkspaceSelectorMenuButton from './WorkspaceSelectorMenuButton';
+import {
+  Organization,
+  Project,
+  PaginationSettings,
+  PaginatedList,
+} from '@bbp/nexus-sdk';
+import { fetchOrgs } from '../../../store/actions/nexus/orgs';
+import { assignOrgAndFetchOrgProjects } from '../../../store/actions/nexus/activeOrg';
+import { FetchableState } from '../../../store/reducers/utils';
+
+export interface WorkspaceSelectorContainerProps {
+  activeOrg: Organization | null;
+  activeProject: Project | null;
+  orgs: FetchableState<PaginatedList<Organization>>;
+  fetchOrgs: (paginationSettings?: PaginationSettings) => void;
+  selectOrg: (orgLabel: string) => void;
+  selectProject: (projectLabel: string) => void;
+  displayPerPage: number;
+}
+
+const WorkspaceSelectorContainer: React.FunctionComponent<
+  WorkspaceSelectorContainerProps
+> = ({ children, ...props }) => <WorkspaceSelectorMenuButton {...props} />;
+
+const mapStateToProps = (state: RootState) => ({
+  displayPerPage: state.uiSettings.pageSizes.workspaceSelectorMenuSize,
+  orgs: (state.nexus && state.nexus.orgs) || {
+    isFetching: false,
+    data: null,
+    error: null,
+  },
+  activeOrg:
+    (state.nexus &&
```

Comment:
These nested checks are so ugly... I'm far from being a proponent of adding yet another dependency, but we may benefit from using [lodash.get](https://www.npmjs.com/package/lodash.get), which is self-contained, to make this easier to read and less brittle.

Reply:
we have a getProp function in the utils folder that can be used for this